frontend/catalyst/ag_primitives.py

```diff
@@ -166,6 +172,9 @@ def _call_catalyst_for(
 ):
     """Dispatch to a Catalyst implementation of for loops."""
 
+    if _emulate_fallback_errors:
+        raise AutoGraphError("Emulated autograph fallback errror")
```

Comment:
I'd like us to find a way to remove the emulation of errors from production code. If we followed this strategy the code base would be littered with such flags.

Reply:
I see your point, but do you think we need fallback here at all?

Reply:
I tried to run your example but it fails for a different reason:
```py
/home/dave/catalyst/frontend/catalyst/ag_primitives.py:382: UserWarning: Tracing of an AutoGraph converted while loop failed with an exception:
  TypeError:    'Undefined' object is not callable
```

Reply:
If this works I'm ok with it, although a realistic test case would be better (unless we can can't come up with one) 👍

Reply:
Yes, I just wrote it here as a draft, I will check it more carefully.

Reply:
Should be implemented by now, @dime10 please check